megamek/src/megamek/client/ui/swing/PlanetaryConditionsDialog.java

```diff
@@ -20,544 +20,618 @@
+            gravTip.append(Messages.getString("PlanetaryConditionsDialog.invalid.number"));
         }
-
-        if ((temper > 200) || (temper < -200)) {
-            JOptionPane
-                    .showMessageDialog(
-                            frame,
-                            Messages
-                                    .getString("PlanetaryConditionsDialog.EnterValidTemperature"),
-                            Messages
-                                    .getString("PlanetaryConditionsDialog.NumberFormatError"),
-                            JOptionPane.ERROR_MESSAGE);
-            return false;
+        if ((grav < 0.1) || (grav > 10.0)) {
+            gravTip.append(Messages.getString("PlanetaryConditionsDialog.invalid.gravRange"));
         }
-
-        int weather = currentWeather = choWeather.getSelectedIndex();
-        if (temper >= 30 && (weather == PlanetaryConditions.WE_LIGHT_HAIL ||
-                weather == PlanetaryConditions.WE_HEAVY_HAIL ||
-                weather == PlanetaryConditions.WE_LIGHT_SNOW || 
-                weather == PlanetaryConditions.WE_SLEET ||
-                weather == PlanetaryConditions.WE_SNOW_FLURRIES ||
-                weather == PlanetaryConditions.WE_HEAVY_SNOW ||
-                weather == PlanetaryConditions.WE_ICE_STORM || 
-                weather == PlanetaryConditions.WE_BLIZZARD ||
-                weather == PlanetaryConditions.WE_MOD_SNOW)) {
-            JOptionPane
-                    .showMessageDialog(
-                        frame, 
-                        Messages.getString("PlanetaryConditionsDialog.EnterValidTemperatureExtreme"), 
-                        Messages.getString("PlanetaryConditionsDialog.NumberFormatError"), 
-                        JOptionPane.ERROR_MESSAGE);
-            return false;
+        
+        int wind = comWind.getSelectedIndex();
+        int atmo = comAtmosphere.getSelectedIndex();
+        
+        if ((chkBlowingSands.isSelected()) && (wind < WI_MOD_GALE) 
+                && (!chkShiftWindStr.isSelected() 
+                        || (conditions.getMinWindStrength() > WI_MOD_GALE) 
+                        || (conditions.getMaxWindStrength() < WI_MOD_GALE))) {
+            windTip.append(Messages.getString("PlanetaryConditionsDialog.invalid.sandsLost"));
+            sandTip.append(Messages.getString("PlanetaryConditionsDialog.invalid.sandsLost"));
         }
 
-        if ((grav < 0.1) || (grav > 10.0)) {
-            JOptionPane
-                    .showMessageDialog(
-                            frame,
-                            Messages
-                                    .getString("PlanetaryConditionsDialog.EnterValidGravity"),
-                            Messages
-                                    .getString("PlanetaryConditionsDialog.NumberFormatError"),
-                            JOptionPane.ERROR_MESSAGE);
-            return false;
+        if ((atmo == ATMO_TRACE) && (wind == WI_LIGHT_GALE)) {
```

Comment:
How about vacuum and wind?

Reply:
Vacuum disables all weather settings, so it doesnt need to be checked there. But it was not actually resetting the weather. Will be corrected.